backend/database-fs.js

```diff
@@ -1,17 +1,16 @@
 'use strict'
 
-import { mkdir, readdir, readFile, unlink, writeFile } from 'node:fs/promises'
-import { join, resolve } from 'node:path'
+import { mkdir, readdir, readFile, rm, unlink, writeFile } from 'node:fs/promises'
+import { dirname, join, resolve } from 'node:path'
 import { checkKey } from '~/shared/domains/chelonia/db.js'
-
-// Initialized in `initStorage()`.
-let dataFolder = ''
+import DatabaseBackend from './DatabaseBackend.js'
 
 // Some operating systems (such as macOS and Windows) use case-insensitive
 // filesystems by default. This can be problematic for Chelonia / Group Income,
 // as we rely on keys being case-sensitive. This is especially relevant for CIDs,
 // where collisions could lead to DoS or data corruption.
-async function testCaseSensitivity () {
+async function testCaseSensitivity (backend: Object) {
+  const { readData, writeData, deleteData } = backend
```

Comment:
This will be my last comment about the binding. It does feel a bit weird, and I think one issue with that could be continually having to double-check (in the future) that this is doing what it should do.